pkg/registry/apis/provisioning/jobs/pullrequest.go

```diff
@@ -114,7 +111,7 @@ func (c *PullRequestCommenter) Process(ctx context.Context, job provisioning.Job
 		return nil
 	}
 
-	logger := c.logger.With("pr", job.Spec.PR)
+	ctx, logger := slogctx.From(ctx, "component", "pull-request-commenter", "pr", job.Spec.PR)
```

Comment:
2 questions on this pattern :) 

- do we really want to log `component` in all the lines in the call stack? 
- wouldn't it be better to log the caller automatically instead of using this `component/connector` fields? See https://cs.opensource.google/go/go/+/go1.23.4:src/log/slog/handler.go;l=130  / #AddSource . I think it would be more maintainable and easy to grasp for others :)